pkgs/applications/editors/emacs/elisp-packages/manual-packages/eaf-map/package.nix

```
@@ -0,0 +1,74 @@
+{
+  # Basic
+  lib,
+  melpaBuild,
+  fetchFromGitHub,
+  # Java Script dependency
+  nodejs,
+  fetchNpmDeps,
+  npmHooks,
+  # Updater
+  nix-update-script,
+}:
+
+melpaBuild (finalAttrs: {
+
+  pname = "eaf-map";
+  version = "0-unstable-2025-07-04";
+
+  src = fetchFromGitHub {
+    owner = "emacs-eaf";
+    repo = "eaf-map";
+    rev = "667865a9422ec71e3518833e1a13806d4f03adfb";
+    hash = "sha256-UgHIzYu/K1NzTDvUn2JkEmiyDEBT9JDmlvp6xG7Nv5k=";
+  };
+
+  env.npmDeps = fetchNpmDeps {
+    name = "${finalAttrs.pname}-npm-deps";
+    inherit (finalAttrs) src;
+    hash = "sha256-prxCFrKvC2dG9BgO3LIKDCFzjn9vFegpvuMy4Eg6Ghs=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    npmHooks.npmConfigHook
+  ];
+
+  postBuild = ''
+    npm run build
+  '';
+
+  files = ''
+    ("*.el"
+     "*.py"
+     "*.js"
+     "src")
+  '';
+
+  postInstall = ''
+    LISPDIR=$out/share/emacs/site-lisp/elpa/${finalAttrs.ename}-${finalAttrs.melpaVersion}
+    cp -r node_modules $LISPDIR/
```

```suggestion
    touch node_modules/.nosearch
    cp -r node_modules $LISPDIR/
```

There are A LOT OF files in this node_modules dir, which slows down Emacs startup A LOT.
Applying this change reduces `emacs-init-time` from 2.3s to 0.3s on my machine.

---

Maybe it is a good idea to add `.nosearch` to other EAF packages.

---

Nice change! It speed up Emacs startup very much, especially when there are many EAF applications. I would like to apply this to other EAF applications.

---

You are right, but the `node_modules` folder is not in the git repository but generated. So the upstream can only change the installation script which is NOT used in Nix derivation. That is, this change in Nix derivation is still necessary.